extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQInsertTest.java

```diff
@@ -99,6 +99,40 @@ public void testInsertOnFoo1()
                      .setExpectedRowSignature(rowSignature)
                      .setExpectedSegment(expectedFooSegments())
                      .setExpectedResultRows(expectedFooRows())
+                     .setExpectedCountersForStageWorkerChannel(
+                         CounterSnapshotBuilder
+                             .with().totalFiles(1)
+                             .buildChannelCounter(),
+                         0, 0, "input0"
+                     )
+                     .setExpectedCountersForStageWorkerChannel(
+                         CounterSnapshotBuilder
+                             .with().rows(5).bytes(251).frames(1)
+                             .buildChannelCounter(),
+                         0, 0, "shuffle"
+                     )
+                     .setExpectedCountersForStageWorkerChannel(
+                         CounterSnapshotBuilder
+                             .with().rows(5).bytes(251).frames(1)
+                             .buildChannelCounter(),
+                         1, 0, "input0"
+                     )
+                     .setExpectedCountersForStageWorkerChannel(
+                         CounterSnapshotBuilder
+                             .with().rows(1, 1, 1, 1, 1).bytes(92, 89, 89, 91, 91).frames(1, 1, 1, 1, 1)
```

Comment:
This kind of thing is _very_ fragile. Make a simple change to the frame format, say, and the byte counts change. Is there a better way to ensure that the counters are updated without creating a fragile test?

Reply:
Also, suppose that things do change and we get a bunch of failures. Is it clear to the poor developer what the new numbers should be? Or, do we have to debug each test, figure out the new numbers, and manually update the code? That will be painful. Is there a better way?

Reply:
Thanks for the suggestion! Yes, it would have been pretty painful to change the bytes in the assertions. I have changed the code so that it no longer asserts for the type of counter unless it is specified. I think this should prevent the tests from being too fragile. Rows and number of frames, if changed, could be calculated without much trial and error.
Please let me know if this looks good or if any other improvements could be made.